src/Microsoft.NetCore.Analyzers/Core/Security/DoNotDisableCertificateValidation.cs

```diff
@@ -0,0 +1,180 @@
+        {
+            return new DiagnosticDescriptor(
+                ruleId,
+                title,
+                message,
+                DiagnosticCategory.Security,
+                DiagnosticHelpers.DefaultDiagnosticSeverity,
+                isEnabledByDefault: false,
+                description: description,
+                helpLinkUri: uri,
+                customTags: WellKnownDiagnosticTags.Telemetry);
+        }
+
+        public sealed override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+
+            // Security analyzer - analyze and report diagnostics on generated code.
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+            
+            context.RegisterCompilationStartAction(
+                (CompilationStartAnalysisContext compilationStartAnalysisContext) =>
+                {
+                    var systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol = compilationStartAnalysisContext.Compilation.GetTypeByMetadataName(
+                            WellKnownTypes.SystemNetSecurityRemoteCertificateValidationCallback);
+                    if (systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol == null)
+                    {
+                        return;
+                    }
+                    
+                    compilationStartAnalysisContext.RegisterOperationAction(
+                        (OperationAnalysisContext operationAnalysisContext) =>
+                        {
+                            var delegateCreationOperation =
+                                (IDelegateCreationOperation)operationAnalysisContext.Operation;
+                            if (systemNetSecurityRemoteCertificateValidationCallbackTypeSymbol.Equals(delegateCreationOperation.Type))
+                            {
+                                var alwaysReturnTrue = false;
+                                var kindOfTargetFunction = delegateCreationOperation.Target.Kind;
+
+                                switch (kindOfTargetFunction)
+                                {
+                                    case OperationKind.AnonymousFunction:
+                                        alwaysReturnTrue = AlwaysReturnTrue(delegateCreationOperation.Target.Descendants());
+                                        break;
+
+                                    case OperationKind.MethodReference:
+                                        var methodReferenceOperation = (IMethodReferenceOperation)delegateCreationOperation.Target;
+                                        var methodSymbol = methodReferenceOperation.Method;
+                                        var blockOperation = methodSymbol.GetTopmostOperationBlock(compilationStartAnalysisContext.Compilation);
```

Comment:
`blockOperation` can be null, for methods declared in metadata/reference assembly. Please add a null check and add unit test(s) for the same.

Reply:
AlwaysReturnTrue() can deal with the situation that blockOpearation is null, when it's null, the countOfReturnStatement equals 0 and the result will be false as expected.

Reply:
The test case is TestDelegatedMethodFromDifferentAssemblyNoDiagnostic

Reply:
you're right. I'll add the null check.

Reply:
Thanks!